### ISSUE TITLE
Dismiss notifications with <ESC> when editor selected.

### ISSUE THREAD

**Issue by terudon** [VSCodeVim/Vim#2258]
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
FEATURE REQUEST

**What happened**:
Pressing <ESC> when editor is focussed (and in normal mode) I can't dismiss notifications.

**What did you expect to happen**:
Without VsCodeVim installed. Pressing <ESC> properly dismisses notifications.
When in normal mode, <ESC> should bubble upwards to trigger dismissal of notifications.

**How to reproduce it (as minimally and precisely as possible)**:
Have a notification visable and have editor focussed. Pressing <ESC> when in normal mode won't dismiss the notification.

**Environment**:
- Extension (VsCodeVim) version: 0.10.6
- VSCode version: 1.19.1
- OS version: Linux Mint 18.1 Serena

**Comment by jpoon** [comment]
I think this may be an upstream issue. Not sure if VS Code offers an extension API to close notifications. Can you file an issue against VS Code and link it here?

**Comment by jpoon** [comment]
Closing this in favor of https://github.com/VSCodeVim/Vim/issues/2258 as the root fix is the same.

**Comment by netpro2k** [comment]
@jpoon I assume you meant to link a different issue number, you pointed back at this issue in your last comment.

**Comment by Chillee** [comment]
@netpro2k Does pressing `shift+<esc>` do what you want?

**Comment by jpoon** [comment]
@netpro2k. Hmm, I actually have no idea what issue I was supposed to link to. Re-opening.

**Comment by terudon** [comment]
@Chillee Yes, shit+<esc> does what I need it to do and is actually a workable solution for me. Thanks!

**Comment by Chillee** [comment]
@jpoon @terudon I think we've discussed this before on another issue, and I think `shift+<esc>` might be the best solution. `<esc>` is an extremely common key that's basically unavoidable in vim, and I don't think forcing the users to close notifications through `<esc>` is the best idea.

I think we should make it more clear that you can use `shift+<esc>`.

Only concern is that I'm not really sure how this plays out with people who use `jj` as `<esc>`.

**Comment by julienvincent** [comment]
FYI, for future readers, I resolved this issue with a keybinding override:

```json
{
  "key": "escape",
  "command": "hideSuggestWidget",
  "when": "suggestWidgetVisible && textInputFocus"
}
```

Which seems to take precedence over the Vim default binding

**Comment by gvnadir** [comment]
I had the same problem and fixed it by adding the lines below to my `keybindings.json` file:

```  
{
  "key": "escape",
  "command": "notifications.clearAll",
  "when": "notificationToastsVisible"
}
```
